integration/poetry_with_version_test.go

```diff
@@ -0,0 +1,92 @@
+package integration_test
```

Comment:
If we wanted to be sure that the version of python selected was the one in `pyproject.toml` then we could do the following:

1. ensure `pyproject.toml` has a version of python that is not the default provided by the `cpython` buildpack. Typically an older version of python would suffice for this.
2. Assert (in the test) that the selected python version came from `pyproject.toml`

However, I don't see other buildpacks for other languages (e.g. `go-mod-vendor` selecting a `go` version from `go-dist`) testing this with integration tests. So I'm not sure we need to establish that pattern here - I think unit tests are sufficient.